packages/create-catalyst/src/index.ts

```diff
@@ -25,80 +26,60 @@ if (!satisfies(process.version, PACKAGE_INFO.engines.node)) {
 
 console.log(chalk.cyanBright(`\n◢ ${PACKAGE_INFO.name} v${PACKAGE_INFO.version}\n`));
 
-const program = new Command();
+export type Options<T> = T extends Command<infer Args, infer Opts> ? [...Args, Opts, T] : never;
 
-const projectName = new Option('--projectName <name>', 'Project name').env('PROJECT_NAME');
-
-const projectDir = new Option('--projectDir <path>', 'Project installation directory').env(
-  'PROJECT_DIR',
-);
-
-const bigCommerceHostname = new Option('--bigCommerceHostname <hostname>', 'BigCommerce Hostname')
-  .default('bigcommerce.com')
-  .env('BIGCOMMERCE_HOSTNAME')
-  .hideHelp();
-
-const sampleDataApiUrl = new Option('--sampleDataApiUrl <url>', 'Sample Data API URL')
-  .default('https://api.bc-sample.store')
-  .env('SAMPLE_DATA_URL')
-  .hideHelp();
-
-const packageManager = new Option('--packageManager <pm>', 'Package manager to use')
-  .env('PACKAGE_MANAGER')
-  .hideHelp();
-
-const ghRef = new Option(
-  '--ghRef <url>',
-  'Clone a specific ref from the bigcommerce/catalyst repository',
-)
-  .default('main')
-  .env('GH_REF')
-  .hideHelp();
-
-const storeHash = new Option('--storeHash <hash>', 'BigCommerce store hash').env(
-  'BIGCOMMERCE_STORE_HASH',
-);
-
-const accessToken = new Option('--accessToken <token>', 'BigCommerce access token').env(
-  'BIGCOMMERCE_ACCESS_TOKEN',
-);
-
-const channelId = new Option('--channelId <id>', 'BigCommerce channel ID').env(
-  'BIGCOMMERCE_CHANNEL_ID',
-);
-
-const customerImpersonationToken = new Option(
-  '--customerImpersonationToken <token>',
-  'BigCommerce customer impersonation token',
-).env('BIGCOMMERCE_CUSTOMER_IMPERSONATION_TOKEN');
```

Comment:
Unfortunately, options must be inlined when using `@commander-js/extra-typings`, otherwise types are inferred as `any`: https://github.com/commander-js/extra-typings?tab=readme-ov-file#usage-tips